src/sentry/integrations/discord/integration.py

```diff
@@ -238,38 +237,24 @@ def _get_discord_user_id(self, auth_code: str, url: str) -> str:
         integration.
 
         """
-        form_data = f"client_id={self.application_id}&client_secret={self.client_secret}&grant_type=authorization_code&code={auth_code}&redirect_uri={url}"
-        headers = {
-            "Content-Type": "application/x-www-form-urlencoded",
-        }
 
         try:
-            response = self.client.post(
-                "https://discord.com/api/v10/oauth2/token",
-                json=False,
-                data=form_data,
-                headers=headers,
-            )
-            token = response["access_token"]  # type: ignore
-
+            access_token = self.client.get_access_token(auth_code, url)
         except ApiError as e:
-            logger.error("discord.install.failed_to_complete_oauth2_flow", extra={"error": str(e)})
-            raise IntegrationError("Failed to complete Discord OAuth2 flow.")
+            logger.exception("discord.install.failed_to_get_access_token", extra={"error": str(e)})
```

Comment:
I think doing both `logger.exception` and the `raise` below it will create 2 Sentry events, do we need that or can the logger still be `error` level?

Reply:
Because it is in an exception, it must be logger.exception and not logger.error. I could just remove the logging statement though.

Reply:
Is it ok to keep the `ApiError as e` and put the str(e) inside the IntegrationError text?